frontend/src/components/scan/utils/RecentScans.jsx

```diff
@@ -94,39 +91,60 @@ RecentScansCard.defaultProps = {
 };
 
 export default function RecentScans({ classification, param }) {
-  // user recent scans
-  const [recentScansUser, LoaderRecentScansUser, refetch] =
-    useAxiosComponentLoader({
-      url: JOB_RECENT_SCANS_USER,
-      method: "POST",
-    });
-  console.debug("recentScansUser", recentScansUser);
+  // api
+  const [
+    loadingScansUser,
+    loadingScans,
+    recentScansUserError,
+    recentScansError,
+    recentScansUser,
+    recentScans,
+    fetchRecentScansUser,
+    fetchRecentscans,
+  ] = useRecentScansStore((state) => [
+    state.loadingScansUser,
+    state.loadingScans,
+    state.recentScansUserError,
+    state.recentScansError,
+    state.recentScansUser,
+    state.recentScans,
+    state.fetchRecentScansUser,
+    state.fetchRecentscans,
+  ]);
 
-  React.useEffect(() => {
-    refetch();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  console.debug(
+    "loadingScansUser",
+    loadingScansUser,
+    "loadingScans",
+    loadingScans,
+  );
 
   // file md5
   const [fileMd5, setFileMd5] = React.useState("");
   if (classification === "file" && param) {
     param.text().then((x) => setFileMd5(md5(x)));
   }
-  // observable/file recent scans
-  const [recentScans, LoaderRecentScans] = useAxiosComponentLoader({
-    url: JOB_RECENT_SCANS,
-    method: "POST",
-    data: { md5: fileMd5.length ? fileMd5 : md5(param) },
-  });
+
+  React.useEffect(() => {
+    fetchRecentScansUser();
+    fetchRecentscans(fileMd5.length ? fileMd5 : md5(param));
+  }, [fetchRecentScansUser, fetchRecentscans, fileMd5, param]);
```

Comment:
I have another request about this: currently we perform to the backend a request for each char inserted, to avoid to flood the backend with a lot of requests (and lot of them are useless: if the user it typing "google.com" we will made some requests for: g, go, goo, etc...) we could try to think about a mechanism to wait the user terminated to write and then perform the request. I'm thinking about something like a delay: wait 1,2 second before send the request and send only one request with all the text. We could implement it ore use a library (the service portal use lodash)